### PR TITLE
Coverage for 'REPAIR' Statements for Hive & Spark3 dialect

### PR DESCRIPTION
**Pull request by R7L208**
This pull request adds code and unit tests for the Hive `MSCK REPAIR` format and Spark `ALTER TABLE REPAIR PARTITIONS` statements.

### Brief summary of the change made

- Add `MsckRepairTableStatementSegment` in Hive dialect
   - Reference this in Spark3 dialect with `get_segment` because it should reflect Hive syntax and, given the need to reflect as Hive updates, to prevent code duplication.
- Update `AlterTableStatementSegment` in Spark3 dialect for Spark syntax equivalent
- Add test cases for all new statement coverage

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_hive.py [hunk 1]
```diff
@@ -556,3 +559,30 @@ class IntervalExpressionSegment(BaseSegment):
...
         ),
     )
+
+
+@hive_dialect.segment()
+class MsckRepairTableStatementSegment(BaseSegment):
+    """A repair table statement.
+
+    `MSCK REPAIR` reference here within Language Manual DDL page:
+    https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL
```
Comment (jpy-git) [inline src/sqlfluff/dialects/dialect_hive.py:569]:
```suggestion
    https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-RecoverPartitions(MSCKREPAIRTABLE)
```
More specific link to the relevant docs

File: src/sqlfluff/dialects/dialect_hive.py [hunk 2]
```diff
@@ -556,3 +559,30 @@ class IntervalExpressionSegment(BaseSegment):
...
+    """
+
+    type = "msck_repair_table_statement"
+
+    match_grammar = Sequence(
+        "MSCK",
+        "REPAIR",
```
Comment (jpy-git) [inline src/sqlfluff/dialects/dialect_hive.py:576]:
```suggestion
        Ref.keyword("REPAIR", optional=True),
```

Comment (jpy-git) [reply src/sqlfluff/dialects/dialect_hive.py:576]: REPAIR is optional in Hive
```
MSCK [REPAIR] TABLE table_name [ADD/DROP/SYNC PARTITIONS];
```

File: src/sqlfluff/dialects/dialect_spark3.py [hunk 3]
```diff
@@ -693,6 +695,18 @@ class DropFunctionStatementSegment(BaseSegment):
...
+@spark3_dialect.segment()
+class MsckRepairTableStatementSegment(hive_dialect.get_segment("MsckRepairTableStatementSegment")):  # type: ignore
+    """A `REPAIR TABLE` statement using Hive MSCK format.
+
+    This class inherits from Hive since Spark leverages Hive format for this command.
+
+    https://spark.apache.org/docs/latest/sql-ref-syntax-ddl-repair-table.html
+    """
+
+    type = "msck_repair_table_statement"
```
Comment (jpy-git) [inline src/sqlfluff/dialects/dialect_spark3.py:707]: REPAIR is not optional in Spark so you'll have to just add the relevant match grammar here and not inherit from hive

File: test/fixtures/dialects/hive/repair_table.sql [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+MSCK REPAIR TABLE table_identifier DROP PARTITIONS;
+MSCK REPAIR TABLE table_identifier SYNC PARTITIONS;
+
+-- REPAIR TABLE with no optional syntax
+MSCK REPAIR TABLE table_identifier;
+
+-- run MSCK REPAIR TABLE to recovers all the partitions
+MSCK REPAIR TABLE t1;
+
+MSCK REPAIR TABLE emp_part DROP PARTITIONS;
```
Comment (jpy-git) [inline test/fixtures/dialects/hive/repair_table.sql:12]: Add some tests cases where the REPAIR keyword isn't there

**Review by jpy-git — COMMENTED** [review]
@R7L208 There's a slight difference between hive and spark so you'll need to make a slight tweak 😄 

FYI we're hoping to get release 0.9.1 out this weekend so let us know if you're able to make the changes before then and we can get this included 🎉

File: src/sqlfluff/dialects/dialect_hive.py (hunk 1 above)
Comment (R7L208) [reply src/sqlfluff/dialects/dialect_hive.py:569]: Hey @jpy-git! `MSCK` just denotes "metastore consistency check". I'm just trying to capture the `REPAIR TABLE` syntax in this segment but can create a separate `MSCK TABLE` for Hive just to capture delta in metadata between the file system and metastore. `MSCK TABLE` is essentially a dry run for `MSCK REPAIR TABLE` in the scenario there's a significant difference then you can use that to adjust cluster-size, etc. for the actual operation.

Let me know your thoughts on this approach :)

File: src/sqlfluff/dialects/dialect_hive.py (hunk 2 above)
Comment (R7L208) [reply src/sqlfluff/dialects/dialect_hive.py:576]: Without repair, this is a different operation and wouldn't fit under the `RepairTableStatementSegment`. Check out comment above but another approach could be to have `MsckTableStatementSegment` for Hive that includes both then just have a 'MsckRepairTableStatement` for Spark.

File: src/sqlfluff/dialects/dialect_spark3.py (hunk 3 above)
Comment (R7L208) [reply src/sqlfluff/dialects/dialect_spark3.py:707]: I will update based on the agreed-upon approach from the above comments.

File: src/sqlfluff/dialects/dialect_hive.py (hunk 1 above)
Comment (jpy-git) [reply src/sqlfluff/dialects/dialect_hive.py:569]: @R7L208 actually let's take your approach and keep the `MsckRepairTableStatementSegment`s as is 👍 

Could you create a separate `MsckTableStatementSegment` statement for the no `REPAIR` case?
And add a note to the hive `MsckTableStatementSegment` docstring noting that there's a separate statement for that case 😄

File: src/sqlfluff/dialects/dialect_hive.py (hunk 2 above)
Comment (R7L208) [reply src/sqlfluff/dialects/dialect_hive.py:576]: I'm personally leaning towards `MsckTableStatementSegment` that captures both in Hive and `MsckRepairTableStatementSegment` in Spark

Comment (R7L208) [reply src/sqlfluff/dialects/dialect_hive.py:582]: Added MsckTableStatementSegment in Hive dialect to address feedback

File: test/fixtures/dialects/hive/repair_table.sql (hunk 4 above)
Comment (R7L208) [reply test/fixtures/dialects/hive/repair_table.sql]: Added `test/fixtures/dialects/five/msck_table.sql`

**Review by jpy-git — APPROVED** [review]
@R7L208 Looks great thanks! LGTM 🚀 ✨

**Pull request merged**